src/snowflake/snowpark/modin/pandas/dataframe.py

```diff
@@ -2355,23 +2355,39 @@ def squeeze(self, axis: Axis | None = None):
                 return Series(query_compiler=self.T._query_compiler)
         return self.copy()
 
-    @dataframe_not_implemented()
-    def stack(self, level=-1, dropna=True):  # noqa: PR01, RT01, D200
+    def stack(
+        self,
+        level: int | str | list = -1,
+        dropna: bool | NoDefault = no_default,
+        sort: bool | NoDefault = no_default,
+        future_stack: bool = False,  # ignored
+    ):
         """
         Stack the prescribed level(s) from columns to index.
         """
         # TODO: SNOW-1063346: Modin upgrade - modin.pandas.DataFrame functions
-        if not isinstance(self.columns, pandas.MultiIndex) or (
-            isinstance(self.columns, pandas.MultiIndex)
-            and is_list_like(level)
-            and len(level) == self.columns.nlevels
+        if future_stack is not False:
+            WarningMessage.ignored_argument(
+                operation="DataFrame.stack",
+                argument="future_stack",
+                message="future_stack parameter has been ignored with Snowflake execution engine",
+            )
+        if dropna is NoDefault:
+            dropna = True
+        if sort is NoDefault:
+            sort = True
+
+        # This ensures that non-pandas MultiIndex objects are caught.
+        is_multiindex = len(self.columns.names) > 1
+        if not is_multiindex or (
+            is_multiindex and is_list_like(level) and len(level) == self.columns.nlevels
```

Comment:
Could we get a case where the user does something like:
`df.stack([0, 0, 0])`?

Reply:
nevermind - looks like the pandas logic has the same "bug(?)":
```python
In [1]: import pandas as pd

In [2]: multicol2 = pd.MultiIndex.from_tuples([('weight', 'kg'),
   ...:                                        ('height', 'm')])
   ...: df_multi_level_cols2 = pd.DataFrame([[1.0, 2.0], [3.0, 4.0]],
   ...:                                     index=['cat', 'dog'],
   ...:                                     columns=multicol2)

In [3]: df_multi_level_cols2.stack([0, 0])
<ipython-input-3-90b21b3f442a>:1: FutureWarning: The previous implementation of stack is deprecated and will be removed in a future version of pandas. See the What's New notes for pandas 2.1.0 for details. Specify future_stack=True to adopt the new implementation and silence this warning.
  df_multi_level_cols2.stack([0, 0])
Out[3]:
cat  height  m     2.0
     weight  kg    1.0
dog  height  m     4.0
     weight  kg    3.0
dtype: float64

In [4]: df_multi_level_cols2.stack([0, 0], future_stack=True)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[4], line 1
----> 1 df_multi_level_cols2.stack([0, 0], future_stack=True)

File ~/.miniconda3/envs/snowpark/lib/python3.10/site-packages/pandas/core/frame.py:9718, in DataFrame.stack(self, level, dropna, sort, future_stack)
   9716         level = [level]
   9717     level = [self.columns._get_level_number(lev) for lev in level]
-> 9718     result = stack_v3(self, level)
   9720 return result.__finalize__(self, method="stack")

File ~/.miniconda3/envs/snowpark/lib/python3.10/site-packages/pandas/core/reshape/reshape.py:897, in stack_v3(frame, level)
    894 if len(level) > 1:
    895     # Arrange columns in the order we want to take them, e.g. level=[2, 0, 1]
    896     sorter = np.argsort(level)
--> 897     ordered_stack_cols = stack_cols._reorder_ilevels(sorter)
    898 else:
    899     ordered_stack_cols = stack_cols

AttributeError: 'Index' object has no attribute '_reorder_ilevels'
```

Reply:
This is just copied from Modin frontend layer, we throw a NotImplementedError anyhow for level being a list

Reply:
Ah I see - my bad!